### PR TITLE
audio: overhaul api for v4

### PR DESCRIPTION
**Pull request by 200sc**

### CODE REVIEW

**Comment by 200sc** [comment]
This should follow #198

File: audio/play.go [hunk 1]
```diff
@@ -1,28 +1,166 @@
...
+// PlayOptions define ways to configure how playback of some audio proceeds
+type PlayOptions struct {
+	FadeOutOnStop time.Duration
+
+	Destination pcm.Writer
+
+	// The span of data that should be copied from reader to writer
+	// at a time. If too low, may lose accuracy on windows. If too high,
+	// may require manual resets when changing audio sources.
+	// Defaults to 125 Milliseconds.
```
Comment (200sc) [inline audio/play.go:41]: TODO: not 125, 50

File: examples/piano/main.go [hunk 2]
```diff
@@ -223,16 +230,50 @@ func main() {
...
+				key.Q: func(gctx context.Context, pitch synth.Pitch) {
+					unison := 4
+					for i := 0; i < unison; i++ {
+						go playWithMonitor(gctx, audio.FadeIn(100*time.Millisecond, audio.LoopReader(src.Saw(synth.AtPitch(pitch)))))
+						go playWithMonitor(gctx, audio.FadeIn(100*time.Millisecond, audio.LoopReader(src.Saw(synth.AtPitch(pitch), synth.Detune(.04)))))
+						go playWithMonitor(gctx, audio.FadeIn(100*time.Millisecond, audio.LoopReader(src.Saw(synth.AtPitch(pitch), synth.Detune(-.05)))))
+					}
+					playWithMonitor(gctx, audio.FadeIn(100*time.Millisecond, audio.LoopReader(src.Saw(synth.AtPitch(pitch)))))
+
+					//playWithMonitor(gctx, audio.FadeIn(100*time.Millisecond, audio.LoopReader(src.Noise())))
```
Comment (Implausiblyfun) [inline examples/piano/main.go:253]: Probably clean this up too?

**Comment by Implausiblyfun** [comment]
lgtm

**Review by Implausiblyfun — APPROVED** [review]

**Pull request merged**